core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildContainerRunner.java

```diff
@@ -36,6 +39,7 @@ public NativeImageBuildContainerRunner(NativeConfig nativeConfig, Path outputDir
 
         outputPath = outputDir == null ? null : outputDir.toAbsolutePath().toString();
         containerName = "build-native-" + RandomStringUtils.random(5, true, false);
+        this.needJava17Image = javaVersion.isJava17OrHigher() == CompiledJavaVersionBuildItem.JavaVersion.Status.TRUE;
```

Comment:
> I wonder if instead of using needJava17Image it would be better (or not) to get the major version and use it directly.

How come?

> Also, although versions between 11 and 17 are not actually supported wouldn't it be better to let them work by using Java 17 for anything higher than 11?

Good point

Reply:
> How come?

Well if we are to support anything other than 11 or 17 it will obviously not work, but if we were to be strict getting the actual version would solve that for us and would also be more future proof, i.e., when we would add support for a new version it would "just work".

Reply:
I think we have a long way to go for that, we can cross that bridge when we reach it :)